asyncscope.md

```diff
@@ -279,201 +352,541 @@ listening activity is bound in the scope of the loop, the lifetime of handling r
+does not change the scope's state---the _`operation-state`_ must be started to close the scope. The scope transitions
+from the closed/joining state to the joined state when the _`join-sender`_ completes. A scope must be in the joined
+state when its destructor starts; otherwise, the destructor invokes `std::terminate()`.
+
+While a scope is open, calls to `nest(snd, scope)` will succeed (unless an exception is thrown by `snd`'s copy- or
+move-constructor while constructing the _`nest-sender`_). Each time a call to `nest(snd, scope)` succeeds, two things
+happen:
+
+1. the scope's count of outstanding senders is incremented before `nest()` returns, and
+2. the given sender, `snd`, is wrapped in a _`nest-sender`_ and returned.
+
+When a call to `nest()` succeeds, the returned _`nest-sender`_ is an associated sender that acts like an RAII handle:
+the scope's internal count is incremented when the sender is created and decremented when the sender is "done with the
+scope", which happens when the sender is destroyed, its _`operation-state`_ is destroyed, or its _`operation-state`_ is
+completed. Moving a _`nest-sender`_ transfers responsibility for decrementing the count from the old instance to the new
+one. Copying a _`nest-sender`_ is permitted if the sender it's wrapping is copyable, but the copy may "fail" since
+copying requires incrementing the scope's count, which is only allowed when the scope is open; if copying fails, the new
+sender is an unassociated sender that behaves as if it were the result of a failed call to `nest()`.
+
+While a scope is closed or joined, calls to `nest(snd, scope)` will always fail by discarding the given sender and
+returning an unassociated _`nest-sender`_. Failed calls to `nest()` do not change the scope's count. Unassociated
+_`nest-senders`_ do not have a reference to the scope they came from and always complete with `stopped` when connected
+and started. Copying or moving an unassociated sender produces another unassociated sender.
+
+The state transitions of a `counting_scope` mean that it can be used to protect asynchronous work from use-after-free
+errors. Given a resource, `res`, and a `counting_scope`, `scope`, obeying the following policy is enough to ensure that
+there are no attempts to use `res` after its lifetime ends:
+
+- all senders that refer to `res` are nested within `scope`; and
+- `scope` is destroyed (and therefore joined) before `res` is destroyed.
+
+Under the standard assumption that the arguments to `nest()` are and remain valid while evaluating `nest()`, it is
+always safe to invoke any supported operation on the returned _`nest-sender`_. Furthermore, if all senders returned from
+`nest()` are eventually started or discarded then the `join()` operation always eventually finishes because the number
+of outstanding senders nested within the corresponding scope is monotonically decreasing. Conversely, "leaking" a
+_`nest-sender`_ will cause the `join()` operation to deadlock.
```

Comment:
What does "leaking" mean in this context? Above you say that if all nest senders are discarded, `join` finishes.